File: onyxsdk-pen/src/androidTest/java/com/onyx/android/sdk/pennative/PenInkResult.java
package com.onyx.android.sdk.pennative;

/** Test-only result pair constructed by the installed Notable JNI library. */
public final class PenInkResult {
    private final PenInk realInk;
    private final PenInk predictionInk;

    public PenInkResult(PenInk realInk, PenInk predictionInk) {
        this.realInk = realInk;
        this.predictionInk = predictionInk;
    }

    public static PenInkResult create(PenInk realInk, PenInk predictionInk) {
        return new PenInkResult(realInk, predictionInk);
    }

    public PenInk getRealInk() { return realInk; }
    public PenInk getPredictionInk() { return predictionInk; }
}

File: onyxsdk-pen/src/main/java/com/onyx/android/sdk/pennative/PenInkResult.kt
package com.onyx.android.sdk.pennative

/** Real and predicted native ink channels compatible with `onyxsdk-pennative:1.0.4`. */
class PenInkResult(
    val realInk: PenInk,
    val predictionInk: PenInk,
) {
    companion object {
        @JvmStatic
        fun create(realInk: PenInk, predictionInk: PenInk): PenInkResult =
            PenInkResult(realInk, predictionInk)
    }
}
